### PR TITLE
Feature: MySQL Flex Server Auto IOPS Scaling

### PR DESCRIPTION
**Pull request by Siddarthyuvaraj**

### CODE REVIEW

File: internal/services/mysql/mysql_flexible_server_resource_test.go [hunk 1]
```diff
@@ -989,6 +990,7 @@ resource "azurerm_mysql_flexible_server" "test" {
     size_gb           = %d
     iops              = %d
     auto_grow_enabled = %t
+	iops_auto_scaling = %t
   }
 }
 `, r.template(data), data.RandomInteger, sizeGB, iops, enabled)
```
Comment (mbfrahry) [inline internal/services/mysql/mysql_flexible_server_resource_test.go:996]: This test is failing as it's missing an argument down below
```suggestion
`, r.template(data), data.RandomInteger, sizeGB, iops, enabled, enabled)
```

File: internal/services/mysql/mysql_flexible_server_resource_test.go [hunk 2]
```diff
@@ -989,6 +990,7 @@ resource "azurerm_mysql_flexible_server" "test" {
     size_gb           = %d
     iops              = %d
     auto_grow_enabled = %t
+	iops_auto_scaling = %t
```
Comment (mbfrahry) [inline internal/services/mysql/mysql_flexible_server_resource_test.go:993]: This isn't lined up properly 
```suggestion
    iops_auto_scaling = %t
```

File: internal/services/mysql/mysql_flexible_server_resource_test.go [hunk 3]
```diff
@@ -704,6 +704,7 @@ resource "azurerm_mysql_flexible_server" "test" {
     size_gb           = 32
     iops              = 400
     auto_grow_enabled = false
+	iops_auto_scaling = false
```
Comment (mbfrahry) [inline internal/services/mysql/mysql_flexible_server_resource_test.go:707]: And here 
```suggestion
    iops_auto_scaling = false
```

**Review by mbfrahry — CHANGES_REQUESTED** [review]
Hey @Siddarthyuvaraj, this looks good but the tests are failing due to missing attributes. I've detailed more down below

**Comment by Siddarthyuvaraj** [comment]
Hi @mbfrahry i have made the changes. Please run the tests and let me know if any further changes required from my end. Thank you.

**Comment by stephybun** [comment]
Thanks for this PR @Siddarthyuvaraj. 

Unfortunately we overlooked another PR that's already been opened to add this property. Since the other PR was opened earlier and is already in active review I'm going to close this one in favour of https://github.com/hashicorp/terraform-provider-azurerm/pull/23329.

Apologies for the confusion and thanks again for your contribution!

**Pull request closed without merging**